protocol/status.go

```diff
@@ -0,0 +1,64 @@
+package protocol
+
+// Status codes (HTTP) as registered with IANA.
+// See: https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
+const (
```

Comment:
Could we use these constants from the standard library? https://golang.org/src/net/http/status.go

Reply:
I have been thinking about this but am not 100% sure if this would be a good idea. The status codes defined here should be compatible with other protocols such as HTTP, gRPC, GraphQL etc. I am not sure if it would be better to create our own status codes independently from HTTP (as gRPC did) or to fully rely on the HTTP status codes. Do you have any input on this?

Reply:
I would like to use codes as for HTTP because it's more clear for end developers. Besides now I see that you are using the same descriptions, nothing specific. So let's use codes from the library and then we can change it (if and when we will need it)

Reply:
Agreed, i will update the PR later today